### ISSUE TITLE
[Deploy Contract] Modify the UX flow for deploying contract

### ISSUE THREAD

**Issue by eswarasai** [MARKETProtocol/dApp#153]
### Description

*Type*: Enhancement

Currently when a user deploys a contract there are three transactions that occur:
- The MarketContract is deployed via the Factory
- The CollateralPool is deployed
- The two are linked via the final transaction

Create a stepwise UX that allows for people to understand this process. When they click deploy, be it a quick deployment or guided deployment, user should be taken to another screen where messaging of each of these steps, their progress, the transaction hash and a link to Etherscan is displayed.

#### We'd recommend to post the suggested mockup before starting work on the UI
Ideally there is a nice grid/list of all the transaction hashes as they occur etc. PFA screenshot of example UI

<img width="1168" alt="screen shot 2018-05-14 at 6 42 38 pm" src="https://user-images.githubusercontent.com/5172086/39999413-aeab2c00-57a6-11e8-86f6-abc9dad79b04.png">

One final note is, we can use this time that a user is waiting for the transactions to be mined to educate them a bit on each contract's function.

### Definition of Done

- [ ] Create needed UX flow for 3 transactions
- [ ] Plan to interact with core team to ensure expectations are met, plan for a few revisions
- [ ] Create ability to neatly display transaction hashes and status of those 3 transacations
- [ ] Create space for some text during each of the three transactions that explains some information on the process/contract being deployed (this can be just place holder `lorem ipsum` for now)
- [ ] Integrate the UX flow with the existing Quick/Guided Deploy Contract flow
- [ ] Add all needed tests for all code written

## Before you `start work`

Please read our contribution [guidelines](https://docs.marketprotocol.io/#contributing) and if there is a bounty involved  please also see [here](https://docs.marketprotocol.io/#gitcoin-and-bounties)

If you have ongoing work from other bounties with us where funding has not been released, please do not pick up a new issue.  We would like to involve as many contributors as possible and parallelize the work flow as much as possible.

Please make sure to comment in the issue here immediately after starting work so we know your plans for implementation and a timeline.

Please also note that in order for work to be accepted, all code must be accompanied by test cases as well.

**Comment by pelsasser** [comment]
This looks good.  Will be interested to see the mockup from whom ever starts work on this.  Ideally their is a nice grid / list of all the transaction hashes as the occur, etc.  

One final note is, we can use this time that a user is waiting for the transactions to be mined to educate them a bit on each contract's function.  

### Definition of Done

- [ ] Create needed UX flow for 3 transactions
- [ ] Create ability to neatly display transaction hashes and status of those 3 transacations
- [ ] Create space for some text during each of the three transactions that explains some information on the process / contract being deployed (this can be just place holder `lorem ipsum` for now)
- [ ] Add all needed tests for all code written

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 0.4 ETH (292.48 USD @ $731.2/ETH) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MARKETProtocol/dApp/153/433).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $7,329.21 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.

 1. [@arun1595](https://gitcoin.co/profile/arun1595) 

 has committed to working on this project to be completed 6 days, 23 hours from now.


__Please answer following questions/comments__ @pelsasser:

No questions for now

**Comment by eswarasai** [comment]
@arun1595 -- Feel free to connect with me or @pelsasser on [Discord](https://marketprotocol.io) channel in case you've any questions w.r.t. the task. Thanks for helping us out again :)

**Comment by bakaoh** [comment]
@eswarasai --- just out of curiosity, why won't we do all these transactions in one factory contract? Something like

```
pragma solidity ^0.4.18;

contract MarketCollateralPool {...}
contract MarketContract {...}

contract Factory {

    function createContract (...) {
        address marketContract = new MarketContract(...);
        address marketCollateralPool = new MarketCollateralPool(marketContract);
        marketContract.transferCreator(msg.sender);
        marketContract.setCollateralPoolContractAddress(marketCollateralPool);
    } 
}
```

It's easier for user to submit fewer transactions.

**Comment by eswarasai** [comment]
@pelsasser -- Thoughts on this?

**Comment by pelsasser** [comment]
@bakaoh @eswarasai 

Originally this was the solution, to deploy all of this via a factory, and that factory than also added contracts to the whitelist.

Currently however, with the size of the oraclize.it API and our contracts, this process runs out of gas and cannot be completed in a single transaction.  I want to get back to attempting something like this, because I still have a few plans on getting the gas costs of deployment down significantly.  

I also believed when I attempted this, the deployment of the factory failed as well, because by itself it was too large to be deployed in a single transaction. 

All that being said, certainly worth taking another try at this. It is definitely a cleaner solution.

**Comment by eswarasai** [comment]
@arun1595 -- Just checking in to see if you've had chance to look into it and any updates on the UI/UX Design/flow you've in mind. Also please let us know your plan of action as well as timeline for this task. Thanks!

**Comment by arun1595** [comment]
@eswarasai I am afraid I won't be able to tackle this issue as I am caught up with other stuff. I'm going to stop work on gitcoin so others from the community can pick this up.

**Comment by eswarasai** [comment]
@arun1595 -- No problem. Thanks for letting us know in advance. Hopefully we'll have you back soon in helping us tackle other open issues :)

**Comment by pelsasser** [comment]
cancelling bounty for a moment for @rojosnow to coordinate further work on this issue.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Cancelled** 

<hr>

__The funding of 0.4 ETH (271.85 USD @ $679.62/ETH) attached to this issue has been **cancelled** by the bounty submitter__

 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $7,940.49 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by alecwesterman** [comment]
I've got this one.

**Comment by eswarasai** [comment]
@alecwesterman -- Hello 👋 
Thanks for showing interest in MARKET Protocol. Feel free to connect with me or @pelsasser on [Discord](https://marketprotocol.io/discord) channel in case you've any questions w.r.t. the task.

**Comment by pelsasser** [comment]
@alecwesterman welcome! Will mark it as in progress!

**Comment by eswarasai** [comment]
@nofxsnap -- Hello 👋 Let me know if you need anything in order to get you started :)

**Comment by pelsasser** [comment]
@nofxsnap - how is this going? any updates on progress here?

**Comment by nofxsnap** [comment]
Hey @pelsasse, I am still digging through code, reading the documentation, and wrapping my head around the dApp. I haven't been able to focus on this issue as much as I had wanted. If I don't make some major progress very soon, I'll open it back up to someone else. Apologies for the delay!

**Comment by pelsasser** [comment]
@nofxsnap no problem! please reach out if we can help with your understanding of the code base or the task at hand.

**Comment by pelsasser** [comment]
@nofxsnap - how is the progress coming here?

**Comment by nofxsnap** [comment]
@pelsasser - Not enough progress to warrant me taking this issue any longer. I'm really sorry, I've been so busy over the past week. I will open this issue back up to the community. I'll try to tackle something else when things settle down and I get more free time.

**Comment by pelsasser** [comment]
thanks for the update @nofxsnap!  

@eswarasai - putting a block on this until we release @marketprotocol/marketprotocol 1.0.0 which moves this to 2 vs 3 transactions.

**Comment by pelsasser** [comment]
closed for #187
